tools/pdp-metadata/pdp-metadata.js

```diff
@@ -37,6 +37,42 @@ async function performCatalogServiceQuery(config, query, variables) {
   return queryResponse.data;
 }
 
+function getJsonLd(product) {
+  const amount = product.priceRange?.minimum?.final?.amount || product.price?.final?.amount;
+  const brand = product.attributes.find((attr) => attr.name === 'brand');
+
+  const schema = {
+    '@context': 'http://schema.org',
+    '@type': 'Product',
+    name: product.name,
+    description: product.meta_description,
+    image: product['og:image'],
+    offers: [],
+    productID: product.sku,
+    sku: product.sku,
+    url: product.path,
+    '@id': product.path,
+  };
+
+  if (brand?.value) {
+    product.brand = {
+      '@type': 'Brand',
+        name: brand?.value,
+    };
+  }
+
+  if (amount?.value && amount?.currency) {
+    schema.offers.push({
```

Comment:
For a `SimpleProductView` we can add a single offer like done here. For a `ComplexProductView` we should add all available variants.

For this we can use a query like this:
```graphql
{
  variants(sku: "MJ12", optionIds: [], pageSize: 100) {
    variants {
      selections
      product {
        sku
        inStock
        __typename
        images(roles: "image") {
          url
        }
        ... on SimpleProductView {
          price {
            ...priceFields
          }
        }
      }
    }
    cursor
  }
}
```

If desired by the customer this could also be limited to specific variant axis (e.g. show color variants, but not the size ones).

Reply:
I don't think this was done in the client-side setJsonLd so far: https://github.com/hlxsites/aem-boilerplate-commerce/blob/e4d9117e66ef15393bb58c145af581a1d02a3fd9/blocks/product-details-custom/product-details-custom.js#L36

Should we update the client side function too?

Reply:
Good point. I think we were actually waiting for the `variants` query to be available on prod CS. If you can re-use the same code, feel free to update PDP as well. Otherwise we can also handle it in a separate issue.